class.jetpack.php

```diff
@@ -5322,25 +5410,161 @@ public static function is_identity_crisis_value_whitelisted( $key, $value ) {
 	}
 
 	/**
+	 * Checks whether the home and siteurl specifically are whitelisted
+	 * Written so that we don't have re-check $key and $value params every time
+	 * we want to check if this site is whitelisted, for example in footer.php
+	 *
+	 * @return bool True = already whitelsisted False = not whitelisted
+	 */
+	public static function jetpack_is_staging_site() {
+		$current_whitelist = Jetpack_Options::get_option( 'identity_crisis_whitelist' );
+		if ( ! $current_whitelist ) {
+			return false;
+		}
+
+		$options_to_check  = Jetpack::identity_crisis_options_to_check();
+		$cloud_options     = Jetpack::init()->get_cloud_site_options( $options_to_check );
+
+		foreach ( $cloud_options as $cloud_key => $cloud_value ) {
+			if ( ! self::is_identity_crisis_value_whitelisted( $cloud_key, $cloud_value ) ) {
+				return false;
+			}
+		}
+		return true;
+	}
+
+	public function identity_crisis_js( $nonce ) {
+?>
+<script>
+(function( $ ) {
+	var SECOND_IN_MS = 500;
```

Comment:
Not sure why I'm getting mentioned here. I can't write Javascript... I do all my transitions via css.